src/checkers/inference/InferenceVisitor.java

```diff
@@ -583,7 +594,7 @@ protected void commonAssignmentCheck(
                                     mono.getCanonicalName(),
                                     mono.getCanonicalName(),
                                     valueType.toString());
-                    return;
+                    return true;
```

Comment:
The `true` is again confusing. Can you add a comment that clarifies?

Reply:
Oh, sorry, this one should be return false because of the check fails. And the code here is outdate with super method and should be updated in a seperate PR. @wmdietl I think I will update the copied code after this update.